sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/TaskKiller.java

```diff
@@ -74,25 +81,44 @@ public static void killTask(TaskID taskId) {
             return;
         }
 
-        synchronized (lock) {
-            LOGGER.info("Enqueueing kill of task: {}", taskId.getValue());
-            tasksToKill.add(taskId);
+        synchronized (LOCK) {
+            TASKS_TO_KILL.add(taskId);
+            LOGGER.info("Enqueued kill of task: {}, {} tasks to kill: {}",
+                    taskId.getValue(),
+                    TASKS_TO_KILL.size(),
+                    TASKS_TO_KILL.stream().map(t -> t.getValue()).collect(Collectors.toList()));
+
+            // Initialize the executor if enabled and not already running.
+            if (executor == null && executorEnabled) {
```

Comment:
Is there any circumstance where there are tasks that need to be killed, but `killTask` won't be called?

Reply:
Not that I'm aware of. The only shortcut would be to invoke the driver directly.